vim-patch:8.1.{1437,2391},8.2.{1255,2185,2294,3033,3034,3037,3038,3043}

janlazo opened the pull request:
Patch v8.2.1255 is required for patch v8.2.3018.
Ref #14865 .

janlazo committed:
vim-patch:8.1.1437: code to handle callbacks is duplicated

Problem:    Code to handle callbacks is duplicated.
Solution:   Add callback_T and functions to deal with it.
https://github.com/vim/vim/commit/3a97bb3f0f8bd118ae23f1c97e55d84ff42eef20

Port Vim's put_callback() as callback_put()
because Neovim's naming convention is {type}_{action},
not {action}_{type}.

Renaming put_callback type as PutCallback.
https://neovim.io/develop/style-guide.xml#Type_Names

janlazo committed:
vim-patch:8.2.1255: cannot use a lambda with quickfix functions

Problem:    Cannot use a lambda with quickfix functions.
Solution:   Add support for lambda. (Yegappan Lakshmanan, closes vim/vim#6499)
https://github.com/vim/vim/commit/d43906d2e5969288f239df851f5ad7b1dc2c7251

bfredl commented on src/nvim/eval/typval.c:
```diff
@@ -1164,6 +1164,35 @@ void callback_free(Callback *callback)
+    tv->v_type = VAR_FUNC;
+    tv->vval.v_string = vim_strsave(cb->data.funcref);
+    func_ref(cb->data.funcref);
+  }
```
what happens when `kCallbackNone` ? is `tv` expected to be in a pre-initialized state (like number 0 or v:null)?

janlazo replied:
I don't know based on the current code. I'll push a new commit to force v:null.

janlazo committed:
vim-patch:8.2.2185: BufUnload is not triggered for the quickfix dummy buffer

Problem:    BufUnload is not triggered for the quickfix dummy buffer.
Solution:   Do trigger BufUnload. (Pontus Leitzler,closes vim/vim#7518, closes vim/vim#7517)
            Fix white space around "=".
https://github.com/vim/vim/commit/1cfb9bb5c06c07f14475f39c4eb57fea1f0dfb69

janlazo committed:
vim-patch:8.2.3033: no error when using alpha delimiter with :global

Problem:    No error when using alpha delimiter with :global.
Solution:   Check the delimiter like with :substitute. (closes vim/vim#8415)
https://github.com/vim/vim/commit/419a40ac9657e39646b2e0f3f71d7736b0c459d1

N/A patches for version.c:

vim-patch:8.1.2391: cannot build when __QNXNTO__ is defined

Problem:    Cannot build when __QNXNTO__ is defined. (Ian Wayne Larson)
Solution:   Move the check for "qansi". (Ken Takata, closes vim/vim#5317)
https://github.com/vim/vim/commit/c95e8d649045add4e77b423bc159dc0d845af559

vim-patch:8.2.2294: VMS: a few remaining problems

Problem:    VMS: a few remaining problems.
Solution:   Add VMS specific changes. Add Lua support. (Zoltan Arpadffy)
https://github.com/vim/vim/commit/82c38fe508155c11a904e6111b5bfb6adde3fb9a

vim-patch:8.2.3034: installing packages on github CI sometimes fails

Problem:    Installing packages on github CI sometimes fails.
Solution:   Update package information first. (Christian Brabandt,
            closes vim/vim#8432)
https://github.com/vim/vim/commit/ef7be8348fd830e409504a442f3f3ad0931cfbc3

vim-patch:8.2.3037: configure reports libcanberra when checking for libsodium

Problem:    Configure reports libcanberra when checking for libsodium.
Solution:   Adjust the message. (Ozaki Kiichi, closes vim/vim#8435)
https://github.com/vim/vim/commit/8ce3ca8961c5968a02de0a0b98f906695331f254

vim-patch:8.2.3038: Amiga built-in version string doesn't include build date

Problem:    Amiga built-in version string doesn't include build date.
Solution:   Add the build date if available. (Ola Söder, closes vim/vim#8437)
https://github.com/vim/vim/commit/cc6504098605f894b557109b618e88913a89914b

vim-patch:8.2.3043: Amiga: cannot get the shell size on MorphOS and AROS

Problem:    Amiga: cannot get the shell size on MorphOS and AROS.
Solution:   Use control sequences. (Ola Söder, closes vim/vim#8438)
https://github.com/vim/vim/commit/d415d269130b233c0f198c75cc159bee721aa55c

janlazo commented:
Patch v8.2.0274 still gets stuck even if it passes on my machine.

kevinhwang91 commented:
I must change the function from `vim.o.qftf = 'v:lua._G.qftf'` to `vim.o.qftf = '{i-> v:lua._G.qftf(i)}'`

janlazo commented:
@kevinhwang91 That workaround shouldn't be necessary. PR with test is welcome.

kevinhwang91 commented:
> @kevinhwang91 That workaround shouldn't be necessary. PR with test is welcome.

https://github.com/neovim/neovim/pull/14909
